src/accordion/accordion.ts

```diff
@@ -175,6 +175,44 @@ export class NgbAccordion implements AfterContentChecked {
     }
   }
 
+  /**
+   * Programmatically expand a panel with a given id if it is closed.
+   */
+  expand(panelId: string) {
+    const panel = this.panels.find(p => p.id === panelId);
+
+    if (panel && !panel.disabled && !panel.isOpen) {
+      this.toggle(panelId);
+    }
+  }
+
+  /**
+   * Programmatically collapse a panel with a given id if it is open.
+   */
+  collapse(panelId: string) {
+    const panel = this.panels.find(p => p.id === panelId);
+
+    if (panel && !panel.disabled && panel.isOpen) {
+      this.toggle(panelId);
+    }
+  }
+
+  /**
+   * Programmatically collapse all panels.
+   */
+  collapseAll() {
+    this.panels.forEach(panel => { panel.isOpen = false; });
```

Comment:
This should rather delegate to the `collapse` call - otherwise we are not taking the `disabled` flag into account. Please add a test for the case assuring that we are not collapsing an open, disabled panel.

Reply:
I wasn't sure if the disabled flag should take effect here or not, because the 'closeOtherPanels' option I believe takes effect even when a panel is disabled (see toggle and _closeOthers).

Reply:
Also, I am not sure how to make a panel in the state of 'open and disabled'.   Is it possible?  If I set disabled on a panel, it closes.   How do I do this?